cmd/estuary-shuttle/rpc.go

```diff
@@ -128,21 +129,17 @@ func (d *Shuttle) addPin(ctx context.Context, contid uint, data cid.Cid, user ui
 		existing := search[0]
 
 		if existing.Failed {
-			// being asked to pin a thing we have marked as failed means the
-			// primary node isnt aware that this pin failed, we need to resend
-			// that notification
-
-			if err := d.sendRpcMessage(ctx, &drpc.Message{
-				Op: drpc.OP_UpdatePinStatus,
-				Params: drpc.MsgParams{
-					UpdatePinStatus: &drpc.UpdatePinStatus{
-						DBID:   contid,
-						Status: types.PinningStatusFailed,
-					},
-				},
-			}); err != nil {
-				log.Errorf("failed to send pin status update: %s", err)
+			// retry a failed existing pin
```

Comment:
no, we do not retry a failed pin. Simply tell Estuary it has failed, and Estuary should mark it (and remove it from consolidation and aggregation zones) and never resend a pin request again.

Reply:
@en0ma but in the case of adding a pin for consolidation, how would we unblock the staging zone from completing consolidation? and in the case i hit, this failure wasn't deterministic - i simply had to get the PinOp code path to run again and it succeeded fine. that makes me think that we need to either make adding a pin more deterministic, or have this type of retry mechanism.

Reply:
> but in the case of adding a pin for consolidation, how would we unblock the staging zone from completing consolidation?

We already addressed this in the last PR (where the worker reruns at x interval, it should always re-check the content location, if all are in the same place, consolidation is done).

> and in the case i hit, this failure wasn't deterministic

just like we do for pin completion, let us handle pin failure - remove this pin from the consolidation set, so that the next time worker runs, it will be aware of this change

Reply:
ok, i can close this PR out and try the failure handling approach. right now that's not in place so if a pin fails during consolidation the bucket does get stuck indefinitely in "consolidating", waiting for the consolidation to finish.